packages/esm-patient-registration-app/src/root.component.tsx

```diff
@@ -1,33 +1,41 @@
 import React from 'react';
 import { BrowserRouter, Route } from 'react-router-dom';
 import { Grid, Row } from 'carbon-components-react';
-import { ExtensionSlot } from '@openmrs/esm-framework';
-import { Resources, ResourcesContext } from './offline.resources';
+import { ExtensionSlot, Session } from '@openmrs/esm-framework';
+import {
+  fetchAddressTemplate,
+  fetchAllRelationshipTypes,
+  fetchCurrentSession,
+  fetchPatientIdentifierTypesWithSources,
+  Resources,
+  ResourcesContext,
+} from './offline.resources';
 import { SavePatientForm } from './patient-registration/form-manager';
 import { PatientRegistration, PatientRegistrationProps } from './patient-registration/patient-registration.component';
+import useSWR from 'swr';
 
 export interface RootProps extends PatientRegistrationProps, Resources {
   savePatientForm: SavePatientForm;
   isOffline: boolean;
 }
 
-export default function Root({
-  currentSession,
-  addressTemplate,
-  relationshipTypes,
-  identifierTypes,
-  savePatientForm,
-  isOffline,
-}: RootProps) {
-  const resources = {
-    currentSession,
-    addressTemplate,
-    relationshipTypes,
-    identifierTypes,
-  };
+export default function Root({ savePatientForm, isOffline }: RootProps) {
+  const { data: currentSession } = useSWR<Session, Error>('patientRegistrationCurrentSessionInfo', fetchCurrentSession);
```

Comment:
Is there some reason for separately fetching the session in the registration? It seems like it would be better to just use the `useSession()` hook from esm-framework.

Reply:
Hi @ibacher! The reason for using the seprate session function is because the session value is cached in here and is then used in the offline functionality of the patient registration page.

I looked through the `useSession` hook and couldn't find, if it's being stored for offline functionality too.

@manuelroemer, can you please help me here.
Thanks!

Reply:
Oh very interesting. Would it be appropriate to make the hook take care of both offline and online mode? That way the api is as clean as possible? What do you guys think?

Reply:
So, rather than have a separate hook to do the whole thing, I think the default pattern is just to tell the service worker to cache the path if the user has opted-in to offline mode as is done [here](https://github.com/openmrs/openmrs-esm-patient-chart/blob/ead9335b6328b9d7bd121605a2b6e0fea11f1af8/packages/esm-form-entry-app/src/index.ts#L25). Then the calls to `useSession()` should still be cached and we reduce the number of calls we're making to the session endpoint.